eclipse-collections-testutils/src/main/java/org/eclipse/collections/impl/test/Verify.java

```diff
@@ -98,11 +97,6 @@ private Verify()
         throw new AssertionError("Suppress default constructor for noninstantiability");
     }
 
-    public static void fail(String message, Throwable cause)
-    {
-        throw new AssertionError(message, cause);
-    }
```

Comment:
I think it makes sense to keep methods like this to preserve source compatibility. We have a few options for the implementation though.

- Delegate to `Assert.fail()` as long as the arguments are compatible. If we go this route, we should mark the method Deprecated and write Javadoc saying that it's inline-able.
- Continue to just throw, but use an Error type from JUnit 5 if the parameters are compatible, or our own subclass of the JUnit 5 error type.
- Unchanged. I think JUnit 5 will work by catching JUnit 4 Error types.